crates/factor-key-value/src/runtime_config.rs

```diff
@@ -1,38 +1,23 @@
 use std::{collections::HashMap, sync::Arc};
 
-use serde::Deserialize;
+use serde::{de::DeserializeOwned, Deserialize};
 use spin_factors::{anyhow, FactorRuntimeConfig};
 use spin_key_value::StoreManager;
 
 #[derive(Deserialize)]
 #[serde(transparent)]
-pub struct RuntimeConfig {
-    pub store_configs: HashMap<String, StoreConfig>,
+pub struct RuntimeConfig<C> {
+    pub store_configs: HashMap<String, C>,
 }
 
-impl FactorRuntimeConfig for RuntimeConfig {
+impl<C: DeserializeOwned> FactorRuntimeConfig for RuntimeConfig<C> {
     const KEY: &'static str = "key_value_store";
 }
 
-#[derive(Deserialize)]
-pub struct StoreConfig {
-    #[serde(rename = "type")]
-    pub type_: String,
-    #[serde(flatten)]
-    pub config: toml::Table,
-}
-
 /// Resolves some piece of runtime configuration to a connection pool
-pub trait RuntimeConfigResolver: Send + Sync {
-    /// Get a store manager for a given store kind and the raw configuration.
-    ///
-    /// `store_kind` is equivalent to the `type` field in the
-    /// `[key_value_store.$storename]` runtime configuration table.
-    fn get_store(
-        &self,
-        store_kind: &str,
-        config: toml::Table,
-    ) -> anyhow::Result<Arc<dyn StoreManager>>;
+pub trait RuntimeConfigResolver<C>: Send + Sync {
```

Comment:
Could `C` be an associated type on this trait instead? e.g. `type StoreConfig: DeserializeOwned`?

Reply:
Oh yea, that's probably better.